app/services/signature-service.js

```diff
@@ -54,7 +54,10 @@ export default class SignatureService extends Service {
     await signSubcase.save();
 
     // Prepare sign flow: create preparation activity and send to SH
-    await uploadPiecesToSigninghub(signFlow, [piece]);
+    const response = await uploadPiecesToSigninghub(signFlow, [piece]);
+    if (!response.ok) {
+      throw new Error('Failed to upload piece to Signing Hub');
+    }
   }
```

Comment:
I'm guessing we want to implement the delete feature here later when this fails.
https://github.com/kanselarij-vlaanderen/frontend-kaleidos/pull/1754.

Could we do this different? Keep the entire process but add a status ('failed to be sent to signinghub') with a button to retry instead of destroying everything and starting over?

Reply:
[KAS-4107](https://kanselarij.atlassian.net/browse/KAS-4107)

Reply:
Will be implementing the delete feature that was just merged

Reply:
added delete when failed